framework/src/Volo.Abp.TextTemplating/Volo/Abp/TextTemplating/TextTemplating/AbpTemplateRenderer.cs

```diff
@@ -0,0 +1,53 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Threading.Tasks;
+using Microsoft.Extensions.DependencyInjection;
+using Microsoft.Extensions.Options;
+using Volo.Abp.DependencyInjection;
+using Volo.Abp.TextTemplating.Scriban;
+
+namespace Volo.Abp.TextTemplating
+{
+    public class AbpTemplateRenderer : ITemplateRenderer, ITransientDependency
+    {
+        protected IServiceProvider ServiceProvider { get; }
+        protected ITemplateDefinitionManager TemplateDefinitionManager { get; }
+        protected AbpTemplateRendererProviderOptions Options { get; }
+
+        public AbpTemplateRenderer(
+            IServiceProvider serviceProvider,
+            ITemplateDefinitionManager templateDefinitionManager,
+            IOptions<AbpTemplateRendererProviderOptions> options)
+        {
+            ServiceProvider = serviceProvider;
+            TemplateDefinitionManager = templateDefinitionManager;
+            Options = options.Value;
+        }
+
+        public virtual async Task<string> RenderAsync(
+            string templateName,
+            object model = null,
+            string cultureName = null,
+            Dictionary<string, object> globalContext = null)
+        {
+            var templateDefinition = TemplateDefinitionManager.Get(templateName);
+
+            var renderEngine = templateDefinition.RenderEngine;
+
+            if (renderEngine.IsNullOrWhiteSpace())
+            {
+                renderEngine = ScribanTemplateRendererProvider.ProviderName;
```

Comment:
Core rendering engine can't be scriban dependent. To set a default engine, define a property (`string DefaultRenderingEngine`) in the text template options class and set it to scriban in the scriban package (force to set without checking the current value). Also, in the razor package set it to razor, but only if it is null. This will make the system backward compatible, while we allow razor to be default only if it is the only engine in the application.